### PR TITLE
watch 3.3.11

### PR DESCRIPTION
**Pull request by yumitsu**
Updated `watch` formula for v3.3.11.

### CODE REVIEW

**Comment by dunn** [comment]
```
==> /usr/local/Cellar/watch/3.3.11/bin/watch --errexit --chgexit --interval 1 date
dyld: Library not loaded: @@HOMEBREW_PREFIX@@/opt/gettext/lib/libintl.8.dylib
  Referenced from: /usr/local/Cellar/watch/3.3.11/bin/watch
  Reason: image not found
Error: watch: failed
```
Looks like an `install_name_tool` error, I think?

**Comment by yumitsu** [comment]
@dunn It seems `watch` loads `libintl` during runtime. Working on it.

**Comment by yumitsu** [comment]
@dunn Hmm, it's still failing. Also, my `install_name_tool` looks like normal:
```
# otool -L `which install_name_tool`
/usr/bin/install_name_tool:
	/usr/lib/libxcselect.dylib (compatibility version 1.0.0, current version 1.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1197.1.1)
```

**Comment by dunn** [comment]
Is it failing locally for you too, then

**Comment by yumitsu** [comment]
@dunn Nope, it's not:
```
brew test --verbose /Users/yumitsu/projects/homebrew/Library/Formula/watch.rb
Testing watch
==> /usr/local/Cellar/watch/3.3.11/bin/watch --errexit --chgexit --interval 1 date
```

**Comment by apjanke** [comment]
Working for me locally too on 10.9.5. Possible issue with the test server? That error is exactly what I see if I remove `gettext` after installing `watch`.

```
[/usr/local/Library/Formula on ⇄ pr-44590]
$ otool -L /usr/local/Cellar/watch/3.3.11/bin/watch
/usr/local/Cellar/watch/3.3.11/bin/watch:
	/usr/lib/libncurses.5.4.dylib (compatibility version 5.4.0, current version 5.4.0)
	/usr/local/opt/gettext/lib/libintl.8.dylib (compatibility version 10.0.0, current version 10.4.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1197.1.1)
[/usr/local/Library/Formula on ⇄ pr-44590]
$ watch --errexit --chgexit --interval 1 date
[/usr/local/Library/Formula on ⇄ pr-44590]
$ brew rm gettext
Uninstalling /usr/local/Cellar/gettext/0.19.6... (1921 files, 22M)
[/usr/local/Library/Formula on ⇄ pr-44590]
$ watch --errexit --chgexit --interval 1 date
dyld: Library not loaded: /usr/local/opt/gettext/lib/libintl.8.dylib
  Referenced from: /usr/local/bin/watch
  Reason: image not found
[1]    6918 trace trap  watch --errexit --chgexit --interval 1 date
```

**Comment by yumitsu** [comment]
@apjanke Not sure. I think it can be an issue with linking process. For now I added option to avoid usage of libintl by using `--disable-nls` option in `configure`. Maybe I should add some notice to `caveats`?

**Comment by dunn** [comment]
If it works for you locally, then it would be a problem with the test bot, and we should figure that out.

**Comment by yumitsu** [comment]
@dunn In that case, should I revert my changes back to to d42f721?

File: Library/Formula/watch.rb [hunk 1]
```diff
@@ -12,15 +18,47 @@ class Watch < Formula
     sha1 "a7c559378bc74cd30d00f962e63d6ee5c705aea1" => :mountain_lion
   end
 
+  depends_on "gettext" => :build
```
Comment (DomT4) [inline Library/Formula/watch.rb]: FYI: This isn't a build-time dep if you're building with Native Language Support. You remove it and runtime breaks.

Comment (apjanke) [inline Library/Formula/watch.rb]: D'oh! Lemme figure out a `test do` case to exercise that.

Comment (apjanke) [inline Library/Formula/watch.rb]: Is the Right Thing to put the `depends_on` inside an `if` block contingent on `--with-nls`? And then put a parallel condition in the `test do`?

Comment (yumitsu) [inline Library/Formula/watch.rb]: @DomT4 `watch` needs `autopoint` from `gettext` package when running `./autogen.sh`, but not when configured with `--disable-nls` flag:
```
autopoint --force $AP_OPTS
if ! grep -q datarootdir po/Makefile.in.in; then
        echo autopoint does not honor dataroot variable, patching.
        sed -i -e 's/^datadir *=\(.*\)/datarootdir = @datarootdir@\
datadir = @datadir@/g' po/Makefile.in.in
fi
```

Comment (DomT4) [inline Library/Formula/watch.rb]: That would become something like:

```ruby
if build.with? "nls"
  depends_on "gettext"
else
  depends_on "gettext" => :build
end
```
But that's clunky and to be honest I'd rather we avoid it by either not making the NLS option available until we resolve the build issues, or resolving the build issues & enabling it by default with no off.

**Comment by bfontaine** [comment]
This PR needs to be rebased on top of `master`.

**Comment by bfontaine** [comment]
Ping?

**Comment by bfontaine** [comment]
Closing due to no response. Feel free to open another PR addressing the comments above. Thanks!

**Pull request closed without merging**